src/validation/extensions/MaxarExtentValidator.ts

```diff
@@ -98,9 +104,637 @@ export class MaxarExtentValidator implements Validator<any> {
+    } else if (geojsonObject.type === "FeatureCollection") {
+      for (const feature of geojsonObject.features || []) {
+        if (!validateGeometry(feature.geometry)) {
+          result = false;
+          break;
+        }
+      }
+    } else {
+      // Direct geometry object
+      result = validateGeometry(geojsonObject);
+    }
+
+    return result;
+  }
+
+  /**
+   * Validates a polygon's coordinates for extent requirements
+   *
+   * @param polygonCoords - The polygon coordinates array
+   * @param path - The path for ValidationIssue instances
+   * @param context - The ValidationContext that any issues will be added to
+   * @returns Whether the polygon meets extent requirements
+   */
+  static validatePolygonExtent(
+    polygonCoords: number[][][],
+    path: string,
+    context: ValidationContext
+  ): boolean {
+    let result = true;
+
+    // Validate each ring (exterior and holes)
+    for (let ringIndex = 0; ringIndex < polygonCoords.length; ringIndex++) {
+      const ring = polygonCoords[ringIndex];
+
+      // Check minimum coordinate count (at least 3 unique coordinates)
+      const uniqueCoords = MaxarExtentValidator.getUniqueCoordinates(ring);
+      if (uniqueCoords.length < 3) {
+        const message = `Extent polygon ring must have at least 3 unique coordinates, but has ${uniqueCoords.length}`;
+        const issue = SemanticValidationIssues.BOUNDING_VOLUMES_INCONSISTENT(
+          path,
+          message
+        );
+        context.addIssue(issue);
+        result = false;
+      }
+
+      // Check for self-intersection
+      if (MaxarExtentValidator.isRingSelfIntersecting(ring)) {
+        const message = `Extent polygon ring ${ringIndex} is self-intersecting, which is forbidden`;
+        const issue = SemanticValidationIssues.BOUNDING_VOLUMES_INCONSISTENT(
+          path,
+          message
+        );
+        context.addIssue(issue);
+        result = false;
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Gets unique coordinates from a ring, excluding the closing coordinate
+   * Uses epsilon-based comparison to handle floating point precision issues
+   *
+   * @param ring - Array of coordinate pairs [longitude, latitude]
+   * @returns Array of unique coordinates with duplicates removed
+   */
+  static getUniqueCoordinates(ring: number[][]): number[][] {
+    if (ring.length === 0) return [];
+
+    const unique: number[][] = [];
+    const epsilon = 1e-10; // Consistent epsilon for coordinate comparison
+
+    // Process all coordinates except the last one (which should close the ring)
+    const coordsToCheck =
+      ring.length > 0 &&
+      MaxarExtentValidator.coordinatesEqual(
+        ring[0],
+        ring[ring.length - 1],
+        epsilon
+      )
+        ? ring.slice(0, -1)
+        : ring;
+
+    for (const coord of coordsToCheck) {
+      // Check if this coordinate is already in the unique list
+      const isDuplicate = unique.some((existingCoord) =>
```

Comment:
Only a remark, no change request: This has O(n²) complexity. This _may_ become an issue for large geometries. But given the epsilon-handling, there is no efficient way for solving that (except for hyper-complex spatial data structures).